app/controllers/AllowancesController.scala

```diff
@@ -46,7 +46,9 @@ class AllowancesController @Inject()(
     implicit request: AuthenticatedRequest[_]): Future[Either[ErrorResponse, GenericViewModel]] =
     extractViewModelWithTaxYear(allowanceService.getAllowances(_))
 
-  override def obtainResult(result: ViewModel)(implicit request: AuthenticatedRequest[_]): Result =
+  override def obtainResult(result: ViewModel)(implicit request: AuthenticatedRequest[_]): Result = {
+    implicit val lang : Lang = request.lang
```

Comment:
This goes for pretty much every occurrence of implicit lang across other controllers too

Can we make an abstract controller or trait to extend from that provides an implicit lang rather than providing it in all these functions?

Reply:
Agreed but We need to have access to "request" to get this lang so we wont be able to move this to a trait or abstract controller .
We wont be needing this once we start working on injecting the views.

Reply:
Can you get a ticket created to DI the views?

Reply:
yes, Working with Tracey for the same. Ticket number is DDCNL-2203.